ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt

```diff
@@ -1,53 +1,171 @@
+ *  * "^" - alias symbol, can be used as follows:
+ *      1. In front of an import path, meaning matching all alias imports from this path, e.g. "^android.*"
+ *      2. In conjunction with "*" operator, meaning matching any alias import - "^*" (<all other alias imports> in IDEA)
+ *  * import paths - these can be full paths, e.g. "java.util.List" as well as wildcard paths, e.g. "kotlin.*"
  *
- * https://developer.android.com/kotlin/style-guide#import_statements
+ * In case the custom property is not provided, the rule defaults to "idea" style.
  */
 class ImportOrderingRule : Rule("import-ordering") {
 
+    private lateinit var importsLayout: String
+    private lateinit var importSorter: ImportSorter
+
+    companion object {
+        /**
+         * Alphabetical with capital letters before lower case letters (e.g. Z before a).
+         * No blank lines between major groups (android, com, junit, net, org, java, javax).
+         * Single group regardless of import type.
+         *
+         * https://developer.android.com/kotlin/style-guide#import_statements
+         */
+        private const val ASCII_PATTERN = "*"
+
+        /**
+         * Default IntelliJ IDEA style: Alphabetical with capital letters before lower case letters (e.g. Z before a),
+         * except such groups as "java", "javax" and "kotlin" that are placed in the end. Within the groups the alphabetical order is preserved.
+         * Alias imports are placed in a separate group in the end of the list with alphabetical order inside.
+         * No blank lines between groups.
+         *
+         * https://github.com/JetBrains/kotlin/blob/ffdab473e28d0d872136b910eb2e0f4beea2e19c/idea/formatter/src/org/jetbrains/kotlin/idea/core/formatter/KotlinCodeStyleSettings.java#L87-L91
+         */
+        private const val IDEA_PATTERN = "*,java.*,javax.*,kotlin.*,^*"
+
+        private const val IDEA_ERROR_MESSAGE = "Imports must be ordered in lexicographic order without any empty lines in-between " +
+            "with \"java\", \"javax\", \"kotlin\" and aliases in the end"
+        private const val ASCII_ERROR_MESSAGE = "Imports must be ordered in lexicographic order without any empty lines in-between"
+        private const val CUSTOM_ERROR_MESSAGE = "Imports must be ordered according to the pattern specified in .editorconfig"
+
+        private val errorMessages = mapOf(
+            IDEA_PATTERN to IDEA_ERROR_MESSAGE,
+            ASCII_PATTERN to ASCII_ERROR_MESSAGE
+        )
+    }
+
     override fun visit(
         node: ASTNode,
         autoCorrect: Boolean,
         emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
     ) {
+        if (node.isRoot()) {
+            val editorConfig = node.getUserData(KtLint.EDITOR_CONFIG_USER_DATA_KEY)!!
+            importsLayout = editorConfig.get("kotlin_imports_layout")
+                ?: "idea" // default to idea in case there's no such entry in .editorconfig
```

Comment:
For Android projects this default violates the style guide https://developer.android.com/kotlin/style-guide#import_statements

Reply:
It violates the style guide, but it complies with how IDEA/Android Studio actually sorts your imports, so I'd say it's a good middle-ground. You can always configure it to "ascii" if you absolutely want to comply with Android style guide.

Also I don't really see a point in defining a style guide, such as you cannot set up Android Studio to comply with it, because there's literally no such capability for Kotlin yet.

Reply:
Probably you could reuse `ANDROID_USER_DATA_KEY` to select default config `ascii` for android projects.

Reply:
I could certainly do that, but I think this will result in 95% of android devs to set it back to `idea`, until Kotlin 1.4 is officially out (I guess only a small percentage is using milestone versions), otherwise it will lead to the same conflicts and frustration as it is now.. I'd rather change it after 1.4 is officially there - what do you guys think?

Reply:
Implemented default to `ascii` in case of android flag enabled